<a href="https://colab.research.google.com/github/Githubprj1/DATA-CLEANING/blob/main/Data_cleaning_movies_data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np

In [2]:
data=pd.read_csv("/content/movies_metadata.csv")

<ipython-input-2-a2e9017387d3>:1: DtypeWarning: Columns (10) have mixed types. Specify dtype option on import or set low_memory=False.
  data=pd.read_csv("/content/movies_metadata.csv")


In [3]:
data.shape

(45466, 24)

45466 - rows
24-columns

In [4]:
data.head()

adult                              belongs_to_collection    budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1  False                                                NaN  65000000   
2  False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3  False                                                NaN  16000000   
4  False  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4                     [{'id': 35, 'name': 'Comedy'}]   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   
3                                   NaN  31357  tt0114885                en   
4                                   NaN  11862  tt0113041                en   

                original_title  \
0                    Toy Story   
1                      Jumanji   
2             Grumpier Old Men   
3            Waiting to Exhale   
4  Father of the Bride Part II   

                                            overview  ... release_date  \
0  Led by Woody, Andy's toys live happily in his ...  ...   1995-10-30   
1  When siblings Judy and Peter discover an encha...  ...   1995-12-15   
2  A family wedding reignites the ancient feud be...  ...   1995-12-22   
3  Cheated on, mistreated and stepped on, the wom...  ...   1995-12-22   
4  Just when George Banks has recovered from his ...  ...   1995-02-10   

       revenue runtime                                   spoken_languages  \
0  373554033.0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0   101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0   127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0   106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title  video vote_average vote_count  
0                    Toy Story  False          7.7     5415.0  
1                      Jumanji  False          6.9     2413.0  
2             Grumpier Old Men  False          6.5       92.0  
3            Waiting to Exhale  False          6.1       34.0  
4  Father of the Bride Part II  False          5.7      173.0  

[5 rows x 24 columns]

In [5]:
data.columns

Index(['adult', 'belongs_to_collection', 'budget', 'genres', 'homepage', 'id',
       'imdb_id', 'original_language', 'original_title', 'overview',
       'popularity', 'poster_path', 'production_companies',
       'production_countries', 'release_date', 'revenue', 'runtime',
       'spoken_languages', 'status', 'tagline', 'title', 'video',
       'vote_average', 'vote_count'],
      dtype='object')

In [6]:
# COLUMN ADULT
data.adult.values   #dataframe.coulmn_name.values - gives the values of the column

array(['False', 'False', 'False', ..., 'False', 'False', 'False'],
      dtype=object)

In [7]:
data.adult.value_counts()

False                                                                                                                             45454
True                                                                                                                                  9
 - Written by Ørnås                                                                                                                   1
 Rune Balot goes to a casino connected to the October corporation to try to wrap up her case once and for all.                        1
 Avalanche Sharks tells the story of a bikini contest that turns into a horrifying affair when it is hit by a shark avalanche.        1
Name: adult, dtype: int64

In [8]:
data.adult.unique()

array(['False', 'True', ' - Written by Ørnås',
       ' Rune Balot goes to a casino connected to the October corporation to try to wrap up her case once and for all.',
       ' Avalanche Sharks tells the story of a bikini contest that turns into a horrifying affair when it is hit by a shark avalanche.'],
      dtype=object)

In [9]:
data.loc[~data.adult.isin(['False','True']),:]

adult  \
19730                                 - Written by Ørnås   
29503   Rune Balot goes to a casino connected to the ...   
35587   Avalanche Sharks tells the story of a bikini ...   

      belongs_to_collection                            budget  \
19730              0.065736  /ff9qCepilowshEtG2GYWwzt2bs4.jpg   
29503              1.931659  /zV8bHuSL6WXoD6FWogP9j4x80bL.jpg   
35587              2.185485  /zaSf5OG7V8X8gqFvly88zDdRm46.jpg   

                                                  genres  \
19730  [{'name': 'Carousel Productions', 'id': 11176}...   
29503  [{'name': 'Aniplex', 'id': 2883}, {'name': 'Go...   
35587  [{'name': 'Odyssey Media', 'id': 17161}, {'nam...   

                                                homepage          id imdb_id  \
19730  [{'iso_3166_1': 'CA', 'name': 'Canada'}, {'iso...  1997-08-20       0   
29503  [{'iso_3166_1': 'US', 'name': 'United States o...  2012-09-29       0   
35587           [{'iso_3166_1': 'CA', 'name': 'Canada'}]  2014-01-01       0   

      original_language                            original_title  overview  \
19730             104.0  [{'iso_639_1': 'en', 'name': 'English'}]  Released   
29503              68.0      [{'iso_639_1': 'ja', 'name': '日本語'}]  Released   
35587              82.0  [{'iso_639_1': 'en', 'name': 'English'}]  Released   

       ... release_date revenue runtime spoken_languages status  tagline  \
19730  ...            1     NaN     NaN              NaN    NaN      NaN   
29503  ...           12     NaN     NaN              NaN    NaN      NaN   
35587  ...           22     NaN     NaN              NaN    NaN      NaN   

       title video vote_average vote_count  
19730    NaN   NaN          NaN        NaN  
29503    NaN   NaN          NaN        NaN  
35587    NaN   NaN          NaN        NaN  

[3 rows x 24 columns]

loc - index based searching in data
where here we select adult column where it is not True or False

Therefore this rows need o be deleted

In [10]:
data = data.loc[data.adult.isin(["False","True"]),:]
data.adult.unique()

array(['False', 'True'], dtype=object)

Now the unwanted data in ADULT column is removed

In [11]:
# changing the values of ADULT COLUMN to numbers
data.adult = data.adult.map({'True':1,'False':0})

In [12]:
data.adult.dtype

dtype('int64')

In [13]:
data.adult.unique()

array([0, 1])

In [14]:
data.head()

adult                              belongs_to_collection    budget  \
0      0  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1      0                                                NaN  65000000   
2      0  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3      0                                                NaN  16000000   
4      0  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4                     [{'id': 35, 'name': 'Comedy'}]   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   
3                                   NaN  31357  tt0114885                en   
4                                   NaN  11862  tt0113041                en   

                original_title  \
0                    Toy Story   
1                      Jumanji   
2             Grumpier Old Men   
3            Waiting to Exhale   
4  Father of the Bride Part II   

                                            overview  ... release_date  \
0  Led by Woody, Andy's toys live happily in his ...  ...   1995-10-30   
1  When siblings Judy and Peter discover an encha...  ...   1995-12-15   
2  A family wedding reignites the ancient feud be...  ...   1995-12-22   
3  Cheated on, mistreated and stepped on, the wom...  ...   1995-12-22   
4  Just when George Banks has recovered from his ...  ...   1995-02-10   

       revenue runtime                                   spoken_languages  \
0  373554033.0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0   101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0   127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0   106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title  video vote_average vote_count  
0                    Toy Story  False          7.7     5415.0  
1                      Jumanji  False          6.9     2413.0  
2             Grumpier Old Men  False          6.5       92.0  
3            Waiting to Exhale  False          6.1       34.0  
4  Father of the Bride Part II  False          5.7      173.0  

[5 rows x 24 columns]

In [15]:
# Belong to Collection column
data.belongs_to_collection.values


array(["{'id': 10194, 'name': 'Toy Story Collection', 'poster_path': '/7G9915LfUQ2lVfwMEEhDsn3kT4B.jpg', 'backdrop_path': '/9FBwqcd9IRruEDUrTdcaafOMKUq.jpg'}",
       nan,
       "{'id': 119050, 'name': 'Grumpy Old Men Collection', 'poster_path': '/nLvUdqgPgm3F85NMCii9gVFUcet.jpg', 'backdrop_path': '/hypTnLot2z8wpFS7qwsQHW1uV8u.jpg'}",
       ..., nan, nan, nan], dtype=object)

In [16]:
#selecting only 1st value
data.belongs_to_collection.values[0]

"{'id': 10194, 'name': 'Toy Story Collection', 'poster_path': '/7G9915LfUQ2lVfwMEEhDsn3kT4B.jpg', 'backdrop_path': '/9FBwqcd9IRruEDUrTdcaafOMKUq.jpg'}"

In [17]:
import ast   # convert string to dictionary
ast.literal_eval(data.belongs_to_collection.values[0])['name']

'Toy Story Collection'

only taking name from the dicitionary becos it is only needed in this column

In [18]:
# here a is nan value so when we compare iot alwas return false
a=np.nan
a == a

False

In [19]:
# we are extracting name from the dictionary of belong to collection column
CollectionNames=[]
for i in data['belongs_to_collection'].values:
  if i == i: # check for values
     CollectionNames.append(ast.literal_eval(i)['name'])
  else:
     CollectionNames.append("Standalone Movie")
     continue


#Collection names is column created fro Belong_to_collection column
data['CollectionName'] = CollectionNames
data=data.drop(columns=['belongs_to_collection'])
data.head()

adult    budget                                             genres  \
0      0  30000000  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1      0  65000000  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2      0         0  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3      0  16000000  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4      0         0                     [{'id': 35, 'name': 'Comedy'}]   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   
3                                   NaN  31357  tt0114885                en   
4                                   NaN  11862  tt0113041                en   

                original_title  \
0                    Toy Story   
1                      Jumanji   
2             Grumpier Old Men   
3            Waiting to Exhale   
4  Father of the Bride Part II   

                                            overview popularity  ...  \
0  Led by Woody, Andy's toys live happily in his ...  21.946943  ...   
1  When siblings Judy and Peter discover an encha...  17.015539  ...   
2  A family wedding reignites the ancient feud be...    11.7129  ...   
3  Cheated on, mistreated and stepped on, the wom...   3.859495  ...   
4  Just when George Banks has recovered from his ...   8.387519  ...   

       revenue runtime                                   spoken_languages  \
0  373554033.0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0   101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0   127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0   106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title  video vote_average vote_count  \
0                    Toy Story  False          7.7     5415.0   
1                      Jumanji  False          6.9     2413.0   
2             Grumpier Old Men  False          6.5       92.0   
3            Waiting to Exhale  False          6.1       34.0   
4  Father of the Bride Part II  False          5.7      173.0   

                   CollectionName  
0            Toy Story Collection  
1                Standalone Movie  
2       Grumpy Old Men Collection  
3                Standalone Movie  
4  Father of the Bride Collection  

[5 rows x 24 columns]

In [20]:
data.CollectionName

0                  Toy Story Collection
1                      Standalone Movie
2             Grumpy Old Men Collection
3                      Standalone Movie
4        Father of the Bride Collection
                      ...              
45461                  Standalone Movie
45462                  Standalone Movie
45463                  Standalone Movie
45464                  Standalone Movie
45465                  Standalone Movie
Name: CollectionName, Length: 45463, dtype: object

In [22]:
#BUDGET
data.budget.values

array(['30000000', '65000000', '0', ..., '0', '0', '0'], dtype=object)

In [23]:
#converting budget type to string to int
data.budget=data.budget.astype('int64')
data.budget.describe()

count    4.546300e+04
mean     4.224579e+06
std      1.742413e+07
min      0.000000e+00
25%      0.000000e+00
50%      0.000000e+00
75%      0.000000e+00
max      3.800000e+08
Name: budget, dtype: float64

In [24]:
sum(data.budget == 0)

36573

36573 budget rows are 0

In [25]:
sum(data.budget == 0)*100/data.shape[0]

80.44563711149726

data.shape[0] = retrieve rows
here 80% of data is missing
It does not mak an sense to impute
we can use some other column to fech this data (or)
we can check imdb site or some other site to find the values (or)
we can delete it -- in most of the cases we shoud not opt the delete options


In [27]:
data.head()

adult    budget                                             genres  \
0      0  30000000  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1      0  65000000  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2      0         0  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3      0  16000000  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4      0         0                     [{'id': 35, 'name': 'Comedy'}]   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   
3                                   NaN  31357  tt0114885                en   
4                                   NaN  11862  tt0113041                en   

                original_title  \
0                    Toy Story   
1                      Jumanji   
2             Grumpier Old Men   
3            Waiting to Exhale   
4  Father of the Bride Part II   

                                            overview popularity  ...  \
0  Led by Woody, Andy's toys live happily in his ...  21.946943  ...   
1  When siblings Judy and Peter discover an encha...  17.015539  ...   
2  A family wedding reignites the ancient feud be...    11.7129  ...   
3  Cheated on, mistreated and stepped on, the wom...   3.859495  ...   
4  Just when George Banks has recovered from his ...   8.387519  ...   

       revenue runtime                                   spoken_languages  \
0  373554033.0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0   101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0   127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0   106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title  video vote_average vote_count  \
0                    Toy Story  False          7.7     5415.0   
1                      Jumanji  False          6.9     2413.0   
2             Grumpier Old Men  False          6.5       92.0   
3            Waiting to Exhale  False          6.1       34.0   
4  Father of the Bride Part II  False          5.7      173.0   

                   CollectionName  
0            Toy Story Collection  
1                Standalone Movie  
2       Grumpy Old Men Collection  
3                Standalone Movie  
4  Father of the Bride Collection  

[5 rows x 24 columns]

In [28]:
#generes
data.genres.values

array(["[{'id': 16, 'name': 'Animation'}, {'id': 35, 'name': 'Comedy'}, {'id': 10751, 'name': 'Family'}]",
       "[{'id': 12, 'name': 'Adventure'}, {'id': 14, 'name': 'Fantasy'}, {'id': 10751, 'name': 'Family'}]",
       "[{'id': 10749, 'name': 'Romance'}, {'id': 35, 'name': 'Comedy'}]",
       ...,
       "[{'id': 28, 'name': 'Action'}, {'id': 18, 'name': 'Drama'}, {'id': 53, 'name': 'Thriller'}]",
       '[]', '[]'], dtype=object)

it is list of dictionary and it is string in pandas

In [29]:
# fetching 1st value
data.genres.values[0]

"[{'id': 16, 'name': 'Animation'}, {'id': 35, 'name': 'Comedy'}, {'id': 10751, 'name': 'Family'}]"

In [30]:
uniqueGenres = set([])

for i in data.genres: #looping through all the rows
    genlist=ast.literal_eval(i)
    for j in genlist:
      uniqueGenres.add(j['name'])

In [31]:
uniqueGenres

{'Action',
 'Adventure',
 'Animation',
 'Comedy',
 'Crime',
 'Documentary',
 'Drama',
 'Family',
 'Fantasy',
 'Foreign',
 'History',
 'Horror',
 'Music',
 'Mystery',
 'Romance',
 'Science Fiction',
 'TV Movie',
 'Thriller',
 'War',
 'Western'}

In [32]:
rows=[[]]
genres=list(uniqueGenres)

for i in data.genres:
    genlist=ast.literal_eval(i)   #converting from string to list
    movieGenres = [x['name'] for x in genlist] #list of all the genres of the movie
    row = []  #store list of 1's and 0's
    for i in genres:
      row.append(int(i in movieGenres))
    rows.append(row)


In [33]:
rows

[[],
 [0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
 [0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
 [0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0],
 [0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 1],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 1],
 [0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0],
 [0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
 [1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1],
 [0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 

In [34]:
genre = pd.DataFrame(rows[1:],columns= genres)
genre

History  Mystery  Animation  Fantasy  Drama  Romance  Music  Foreign  \
0            0        0          1        0      0        0      0        0   
1            0        0          0        1      0        0      0        0   
2            0        0          0        0      0        1      0        0   
3            0        0          0        0      1        1      0        0   
4            0        0          0        0      0        0      0        0   
...        ...      ...        ...      ...    ...      ...    ...      ...   
45458        0        0          0        0      1        0      0        0   
45459        0        0          0        0      1        0      0        0   
45460        0        0          0        0      1        0      0        0   
45461        0        0          0        0      0        0      0        0   
45462        0        0          0        0      0        0      0        0   

       Family  Science Fiction  Western  Crime  Action  TV Movie  Comedy  War  \
0           1                0        0      0       0         0       1    0   
1           1                0        0      0       0         0       0    0   
2           0                0        0      0       0         0       1    0   
3           0                0        0      0       0         0       1    0   
4           0                0        0      0       0         0       1    0   
...       ...              ...      ...    ...     ...       ...     ...  ...   
45458       1                0        0      0       0         0       0    0   
45459       0                0        0      0       0         0       0    0   
45460       0                0        0      0       1         0       0    0   
45461       0                0        0      0       0         0       0    0   
45462       0                0        0      0       0         0       0    0   

       Horror  Thriller  Documentary  Adventure  
0           0         0            0          0  
1           0         0            0          1  
2           0         0            0          0  
3           0         0            0          0  
4           0         0            0          0  
...       ...       ...          ...        ...  
45458       0         0            0          0  
45459       0         0            0          0  
45460       0         1            0          0  
45461       0         0            0          0  
45462       0         0            0          0  

[45463 rows x 20 columns]

In [35]:
data = pd.concat([data,genre], axis=1)  # joining two dataframe DATA AND GENRE , axis=1 means join columns
data.shape,genre.shape

((45466, 44), (45463, 20))

In [36]:
data.shape

(45466, 44)

In [37]:
data

adult      budget                                             genres  \
0        0.0  30000000.0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1        0.0  65000000.0  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2        0.0         0.0  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3        0.0  16000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4        0.0         0.0                     [{'id': 35, 'name': 'Comedy'}]   
...      ...         ...                                                ...   
45464    0.0         0.0                                                 []   
45465    0.0         0.0                                                 []   
19730    NaN         NaN                                                NaN   
29503    NaN         NaN                                                NaN   
35587    NaN         NaN                                                NaN   

                                   homepage      id    imdb_id  \
0      http://toystory.disney.com/toy-story     862  tt0114709   
1                                       NaN    8844  tt0113497   
2                                       NaN   15602  tt0113228   
3                                       NaN   31357  tt0114885   
4                                       NaN   11862  tt0113041   
...                                     ...     ...        ...   
45464                                   NaN  227506  tt0008536   
45465                                   NaN  461257  tt6980792   
19730                                   NaN     NaN        NaN   
29503                                   NaN     NaN        NaN   
35587                                   NaN     NaN        NaN   

      original_language               original_title  \
0                    en                    Toy Story   
1                    en                      Jumanji   
2                    en             Grumpier Old Men   
3                    en            Waiting to Exhale   
4                    en  Father of the Bride Part II   
...                 ...                          ...   
45464                en          Satana likuyushchiy   
45465                en                     Queerama   
19730               NaN                          NaN   
29503               NaN                          NaN   
35587               NaN                          NaN   

                                                overview popularity  ...  \
0      Led by Woody, Andy's toys live happily in his ...  21.946943  ...   
1      When siblings Judy and Peter discover an encha...  17.015539  ...   
2      A family wedding reignites the ancient feud be...    11.7129  ...   
3      Cheated on, mistreated and stepped on, the wom...   3.859495  ...   
4      Just when George Banks has recovered from his ...   8.387519  ...   
...                                                  ...        ...  ...   
45464  In a small town live two brothers, one a minis...   0.003503  ...   
45465  50 years after decriminalisation of homosexual...   0.163015  ...   
19730                                                NaN        NaN  ...   
29503                                                NaN        NaN  ...   
35587                                                NaN        NaN  ...   

      Western Crime Action TV Movie  Comedy  War Horror Thriller Documentary  \
0         0.0   0.0    0.0      0.0     1.0  0.0    0.0      0.0         0.0   
1         0.0   0.0    0.0      0.0     0.0  0.0    0.0      0.0         0.0   
2         0.0   0.0    0.0      0.0     1.0  0.0    0.0      0.0         0.0   
3         0.0   0.0    0.0      0.0     1.0  0.0    0.0      0.0         0.0   
4         0.0   0.0    0.0      0.0     1.0  0.0    0.0      0.0         0.0   
...       ...   ...    ...      ...     ...  ...    ...      ...         ...   
45464     NaN   NaN    NaN      NaN     NaN  NaN    NaN      NaN         NaN   
45465     NaN   NaN    NaN      NaN    

In [38]:
data.columns

Index(['adult', 'budget', 'genres', 'homepage', 'id', 'imdb_id',
       'original_language', 'original_title', 'overview', 'popularity',
       'poster_path', 'production_companies', 'production_countries',
       'release_date', 'revenue', 'runtime', 'spoken_languages', 'status',
       'tagline', 'title', 'video', 'vote_average', 'vote_count',
       'CollectionName', 'History', 'Mystery', 'Animation', 'Fantasy', 'Drama',
       'Romance', 'Music', 'Foreign', 'Family', 'Science Fiction', 'Western',
       'Crime', 'Action', 'TV Movie', 'Comedy', 'War', 'Horror', 'Thriller',
       'Documentary', 'Adventure'],
      dtype='object')

In [39]:
len(genres)

20

In [40]:
data=data.drop(columns=['genres'])

In [41]:
data.head()

adult      budget                              homepage     id    imdb_id  \
0    0.0  30000000.0  http://toystory.disney.com/toy-story    862  tt0114709   
1    0.0  65000000.0                                   NaN   8844  tt0113497   
2    0.0         0.0                                   NaN  15602  tt0113228   
3    0.0  16000000.0                                   NaN  31357  tt0114885   
4    0.0         0.0                                   NaN  11862  tt0113041   

  original_language               original_title  \
0                en                    Toy Story   
1                en                      Jumanji   
2                en             Grumpier Old Men   
3                en            Waiting to Exhale   
4                en  Father of the Bride Part II   

                                            overview popularity  \
0  Led by Woody, Andy's toys live happily in his ...  21.946943   
1  When siblings Judy and Peter discover an encha...  17.015539   
2  A family wedding reignites the ancient feud be...    11.7129   
3  Cheated on, mistreated and stepped on, the wom...   3.859495   
4  Just when George Banks has recovered from his ...   8.387519   

                        poster_path  ... Western Crime Action  TV Movie  \
0  /rhIRbceoE9lR4veEXuwCC2wARtG.jpg  ...     0.0   0.0    0.0       0.0   
1  /vzmL6fP7aPKNKPRTFnZmiUfciyV.jpg  ...     0.0   0.0    0.0       0.0   
2  /6ksm1sjKMFLbO7UY2i6G1ju9SML.jpg  ...     0.0   0.0    0.0       0.0   
3  /16XOMpEaLWkrcPqSQqhTmeJuqQl.jpg  ...     0.0   0.0    0.0       0.0   
4  /e64sOI48hQXyru7naBFyssKFxVd.jpg  ...     0.0   0.0    0.0       0.0   

   Comedy  War Horror Thriller Documentary Adventure  
0     1.0  0.0    0.0      0.0         0.0       0.0  
1     0.0  0.0    0.0      0.0         0.0       1.0  
2     1.0  0.0    0.0      0.0         0.0       0.0  
3     1.0  0.0    0.0      0.0         0.0       0.0  
4     1.0  0.0    0.0      0.0         0.0       0.0  

[5 rows x 43 columns]

In [42]:
data.columns

Index(['adult', 'budget', 'homepage', 'id', 'imdb_id', 'original_language',
       'original_title', 'overview', 'popularity', 'poster_path',
       'production_companies', 'production_countries', 'release_date',
       'revenue', 'runtime', 'spoken_languages', 'status', 'tagline', 'title',
       'video', 'vote_average', 'vote_count', 'CollectionName', 'History',
       'Mystery', 'Animation', 'Fantasy', 'Drama', 'Romance', 'Music',
       'Foreign', 'Family', 'Science Fiction', 'Western', 'Crime', 'Action',
       'TV Movie', 'Comedy', 'War', 'Horror', 'Thriller', 'Documentary',
       'Adventure'],
      dtype='object')

In [43]:
data.shape

(45466, 43)

In [44]:
#HOMEPAGE Column
data.homepage.values

array(['http://toystory.disney.com/toy-story', nan, nan, ..., nan, nan,
       nan], dtype=object)

In [45]:
data.homepage.values[0]

'http://toystory.disney.com/toy-story'

In [46]:
data.homepage.isnull() #true- null present, false-not null value

0        False
1         True
2         True
3         True
4         True
         ...  
45464     True
45465     True
19730     True
29503     True
35587     True
Name: homepage, Length: 45466, dtype: bool

In [47]:
#creating DERIVED COLUMN
data["homePagePresent"] = (~data.homepage.isnull())

so in derived column - homepagepresent , if it is null value - make as True and not null value - make as False

In [48]:
data.homePagePresent.values

array([ True, False, False, ..., False, False, False])

In [49]:
#dropping the homepage
data = data.drop(columns = ["homepage"])

In [50]:
data.homePagePresent.dtype

dtype('bool')

In [51]:
data

adult      budget      id    imdb_id original_language  \
0        0.0  30000000.0     862  tt0114709                en   
1        0.0  65000000.0    8844  tt0113497                en   
2        0.0         0.0   15602  tt0113228                en   
3        0.0  16000000.0   31357  tt0114885                en   
4        0.0         0.0   11862  tt0113041                en   
...      ...         ...     ...        ...               ...   
45464    0.0         0.0  227506  tt0008536                en   
45465    0.0         0.0  461257  tt6980792                en   
19730    NaN         NaN     NaN        NaN               NaN   
29503    NaN         NaN     NaN        NaN               NaN   
35587    NaN         NaN     NaN        NaN               NaN   

                    original_title  \
0                        Toy Story   
1                          Jumanji   
2                 Grumpier Old Men   
3                Waiting to Exhale   
4      Father of the Bride Part II   
...                            ...   
45464          Satana likuyushchiy   
45465                     Queerama   
19730                          NaN   
29503                          NaN   
35587                          NaN   

                                                overview popularity  \
0      Led by Woody, Andy's toys live happily in his ...  21.946943   
1      When siblings Judy and Peter discover an encha...  17.015539   
2      A family wedding reignites the ancient feud be...    11.7129   
3      Cheated on, mistreated and stepped on, the wom...   3.859495   
4      Just when George Banks has recovered from his ...   8.387519   
...                                                  ...        ...   
45464  In a small town live two brothers, one a minis...   0.003503   
45465  50 years after decriminalisation of homosexual...   0.163015   
19730                                                NaN        NaN   
29503                                                NaN        NaN   
35587                                                NaN        NaN   

                            poster_path  \
0      /rhIRbceoE9lR4veEXuwCC2wARtG.jpg   
1      /vzmL6fP7aPKNKPRTFnZmiUfciyV.jpg   
2      /6ksm1sjKMFLbO7UY2i6G1ju9SML.jpg   
3      /16XOMpEaLWkrcPqSQqhTmeJuqQl.jpg   
4      /e64sOI48hQXyru7naBFyssKFxVd.jpg   
...                                 ...   
45464  /aorBPO7ak8e8iJKT5OcqYxU3jlK.jpg   
45465  /s5UkZt6NTsrS7ZF0Rh8nzupRlIU.jpg   
19730                               NaN   
29503                               NaN   
35587                               NaN   

                                    production_companies  ... Crime Action  \
0         [{'name': 'Pixar Animation Studios', 'id': 3}]  ...   0.0    0.0   
1      [{'name': 'TriStar Pictures', 'id': 559}, {'na...  ...   0.0    0.0   
2      [{'name': 'Warner Bros.', 'id': 6194}, {'name'...  ...   0.0    0.0   
3      [{'name': 'Twentieth Century Fox Film Corporat...  ...   0.0    0.0   
4      [{'name': 'Sandollar Productions', 'id': 5842}...  ...   0.0    0.0   
...                                                  ...  ...   ...    ...   
45464               [{'name': 'Yermoliev', 'id': 88753}]  ...   NaN    NaN   
45465                                                 []  ...   NaN    NaN   
19730                                                NaN  ...   0.0    0.0   
29503                                                NaN  ...   0.0    0.0   
35587                                                NaN  ...   0.0    0.0   

       TV Movie  Comedy  War Horror Thriller Documentary Adventure  \
0           0.0     1.0  0.0    0.0      0.0         0.0       0.0   
1           0.0     0.0  0.0    0.0      0.0         0.0       1.0   
2           0.0     1.0  0.0    0.0      0.0         0.0       0.0   
3           0.0     1.0  0.0    0.0      0.0         0.0       0.0   
4           0.0     1.0  0.0    0.0      0.0         0.0       0.0   
...         ...     ...  ...    ...      .

In [52]:
# converting the bool to numbers
data.homePagePresent = data.homePagePresent.map({True:1,False:0})

In [53]:
data.homePagePresent.value_counts

<bound method IndexOpsMixin.value_counts of 0        1
1        0
2        0
3        0
4        0
        ..
45464    0
45465    0
19730    0
29503    0
35587    0
Name: homePagePresent, Length: 45466, dtype: int64>

In [54]:
data

adult      budget      id    imdb_id original_language  \
0        0.0  30000000.0     862  tt0114709                en   
1        0.0  65000000.0    8844  tt0113497                en   
2        0.0         0.0   15602  tt0113228                en   
3        0.0  16000000.0   31357  tt0114885                en   
4        0.0         0.0   11862  tt0113041                en   
...      ...         ...     ...        ...               ...   
45464    0.0         0.0  227506  tt0008536                en   
45465    0.0         0.0  461257  tt6980792                en   
19730    NaN         NaN     NaN        NaN               NaN   
29503    NaN         NaN     NaN        NaN               NaN   
35587    NaN         NaN     NaN        NaN               NaN   

                    original_title  \
0                        Toy Story   
1                          Jumanji   
2                 Grumpier Old Men   
3                Waiting to Exhale   
4      Father of the Bride Part II   
...                            ...   
45464          Satana likuyushchiy   
45465                     Queerama   
19730                          NaN   
29503                          NaN   
35587                          NaN   

                                                overview popularity  \
0      Led by Woody, Andy's toys live happily in his ...  21.946943   
1      When siblings Judy and Peter discover an encha...  17.015539   
2      A family wedding reignites the ancient feud be...    11.7129   
3      Cheated on, mistreated and stepped on, the wom...   3.859495   
4      Just when George Banks has recovered from his ...   8.387519   
...                                                  ...        ...   
45464  In a small town live two brothers, one a minis...   0.003503   
45465  50 years after decriminalisation of homosexual...   0.163015   
19730                                                NaN        NaN   
29503                                                NaN        NaN   
35587                                                NaN        NaN   

                            poster_path  \
0      /rhIRbceoE9lR4veEXuwCC2wARtG.jpg   
1      /vzmL6fP7aPKNKPRTFnZmiUfciyV.jpg   
2      /6ksm1sjKMFLbO7UY2i6G1ju9SML.jpg   
3      /16XOMpEaLWkrcPqSQqhTmeJuqQl.jpg   
4      /e64sOI48hQXyru7naBFyssKFxVd.jpg   
...                                 ...   
45464  /aorBPO7ak8e8iJKT5OcqYxU3jlK.jpg   
45465  /s5UkZt6NTsrS7ZF0Rh8nzupRlIU.jpg   
19730                               NaN   
29503                               NaN   
35587                               NaN   

                                    production_companies  ... Crime Action  \
0         [{'name': 'Pixar Animation Studios', 'id': 3}]  ...   0.0    0.0   
1      [{'name': 'TriStar Pictures', 'id': 559}, {'na...  ...   0.0    0.0   
2      [{'name': 'Warner Bros.', 'id': 6194}, {'name'...  ...   0.0    0.0   
3      [{'name': 'Twentieth Century Fox Film Corporat...  ...   0.0    0.0   
4      [{'name': 'Sandollar Productions', 'id': 5842}...  ...   0.0    0.0   
...                                                  ...  ...   ...    ...   
45464               [{'name': 'Yermoliev', 'id': 88753}]  ...   NaN    NaN   
45465                                                 []  ...   NaN    NaN   
19730                                                NaN  ...   0.0    0.0   
29503                                                NaN  ...   0.0    0.0   
35587                                                NaN  ...   0.0    0.0   

       TV Movie  Comedy  War Horror Thriller Documentary Adventure  \
0           0.0     1.0  0.0    0.0      0.0         0.0       0.0   
1           0.0     0.0  0.0    0.0      0.0         0.0       1.0   
2           0.0     1.0  0.0    0.0      0.0         0.0       0.0   
3           0.0     1.0  0.0    0.0      0.0         0.0       0.0   
4           0.0     1.0  0.0    0.0      0.0         0.0       0.0   
...         ...     ...  ...    ...      .

In [55]:
data.columns

Index(['adult', 'budget', 'id', 'imdb_id', 'original_language',
       'original_title', 'overview', 'popularity', 'poster_path',
       'production_companies', 'production_countries', 'release_date',
       'revenue', 'runtime', 'spoken_languages', 'status', 'tagline', 'title',
       'video', 'vote_average', 'vote_count', 'CollectionName', 'History',
       'Mystery', 'Animation', 'Fantasy', 'Drama', 'Romance', 'Music',
       'Foreign', 'Family', 'Science Fiction', 'Western', 'Crime', 'Action',
       'TV Movie', 'Comedy', 'War', 'Horror', 'Thriller', 'Documentary',
       'Adventure', 'homePagePresent'],
      dtype='object')

In [56]:
#ID column
data.id.values

array(['862', '8844', '15602', ..., nan, nan, nan], dtype=object)

data type is string . It shoud be a integer

In [57]:
#id should be unique
#finding duplicate value - total row -  id column unique value
(data.shape[0])- len(data.id.unique())  #32 duplicate values are there

32

In [58]:
#to delete id column
data = data.drop_duplicates(subset = 'id')  # delete the rows which has all the id column value same

In [59]:
data.shape

(45434, 43)

In [61]:
(data.shape[0])- len(data.id.unique())  #now no unique id values

0

In [62]:
sum(data.id.isnull()) #one null value

1

In [63]:
#dropping the null value in id column
data = data.dropna(subset = ['id'])

In [64]:
data.id.values

array(['862', '8844', '15602', ..., '67758', '227506', '461257'],
      dtype=object)

In [65]:
#convert string to int
data.id=data.id.astype('int64')

In [66]:
data.id.values.dtype

dtype('int64')

# IMDB ID COLUMN

In [68]:
#imdb_id
data.imdb_id.values

array(['tt0114709', 'tt0113497', 'tt0113228', ..., 'tt0303758',
       'tt0008536', 'tt6980792'], dtype=object)

In [69]:
(data.shape[0])- len(data.imdb_id.unique())  # finding duplicates (total row - unique value of imdb column)

16

In [70]:
data = data.drop_duplicates(subset = 'imdb_id')

In [72]:
(data.shape[0])- len(data.imdb_id.unique())

0

In [75]:
sum(data.imdb_id.isnull())

1

In [76]:
#dropping the null value in imdb column
data = data.dropna(subset = ['imdb_id'])

In [77]:
sum(data.imdb_id.isnull())

0

In [78]:
data.shape

(45416, 43)

# orginal column


In [79]:
#Original column
data.original_language.values

array(['en', 'en', 'en', ..., 'en', 'en', 'en'], dtype=object)

In [80]:
data.original_language.values[0]  #1st value

'en'

In [81]:
len(data.original_language.unique())  #90 unique values

90

In [82]:
data.original_language.value_counts()  # gives counts of each unique values

en    32238
fr     2435
it     1529
ja     1346
de     1079
      ...  
ay        1
rw        1
cy        1
tg        1
si        1
Name: original_language, Length: 89, dtype: int64

In [83]:
sum(data.original_language.isnull()) #null or missing values

11

In [84]:
# 11 null values so dropping the values
data = data.dropna(subset=['original_language'])

#Title and Orginal title

In [85]:
# Original_title and title are going to give same data so e are selecting which has high unique values.
data.title.isnull().sum()  #give sum of nan values

3

In [86]:
#title column
len(data.title.unique())*100/data.shape[0]  #percentage of unique values in title column

93.05803325624932

In [87]:
data.original_title.isnull().sum()

0

In [88]:
len(data.original_title.unique())*100/data.shape[0]

95.46305472965533

In [89]:
#ORIGINAL TITLE HAS HIGH UNIQUE VALUES AND NO NUL VALUES
#therefore dropping the title column
data = data.drop(columns = ['title'])

In [90]:
#always make sring to lower or upper case
data.original_title = data['original_title'].str.lower()

In [91]:
data.original_title.values

array(['toy story', 'jumanji', 'grumpier old men', ..., 'betrayal',
       'satana likuyushchiy', 'queerama'], dtype=object)

In [92]:
data.original_title = data['original_title'].str.strip()  #DELETE UNWANTED SPACES

In [93]:
data.original_title

0                          toy story
1                            jumanji
2                   grumpier old men
3                  waiting to exhale
4        father of the bride part ii
                    ...             
45461                        رگ خواب
45462            siglo ng pagluluwal
45463                       betrayal
45464            satana likuyushchiy
45465                       queerama
Name: original_title, Length: 45405, dtype: object

#Overview
HERE WE HAVE MANY DIFFERENT LANGUAGE WHICH CAN BE HANDLED IN NLP

In [94]:
#overview
data.overview.values

array(["Led by Woody, Andy's toys live happily in his room until Andy's birthday brings Buzz Lightyear onto the scene. Afraid of losing his place in Andy's heart, Woody plots against Buzz. But when circumstances separate Buzz and Woody from their owner, the duo eventually learns to put aside their differences.",
       "When siblings Judy and Peter discover an enchanted board game that opens the door to a magical world, they unwittingly invite Alan -- an adult who's been trapped inside the game for 26 years -- into their living room. Alan's only hope for freedom is to finish the game, which proves risky as all three find themselves running from giant rhinoceroses, evil monkeys and other terrifying creatures.",
       "A family wedding reignites the ancient feud between next-door neighbors and fishing buddies John and Max. Meanwhile, a sultry Italian divorcée opens a restaurant at the local bait shop, alarming the locals who worry she'll scare the fish away. But she's less interested in

In [95]:
data.overview = data['overview'].str.lower()

In [96]:
data.overview = data['overview'].str.strip()

In [97]:
data.overview

0        led by woody, andy's toys live happily in his ...
1        when siblings judy and peter discover an encha...
2        a family wedding reignites the ancient feud be...
3        cheated on, mistreated and stepped on, the wom...
4        just when george banks has recovered from his ...
                               ...                        
45461          rising and falling between a man and woman.
45462    an artist struggles to finish his work while a...
45463    when one of her hits goes wrong, a professiona...
45464    in a small town live two brothers, one a minis...
45465    50 years after decriminalisation of homosexual...
Name: overview, Length: 45405, dtype: object

In [98]:
sum(data.overview.isnull()) #951 null values

951

In [99]:
#imputing null values as 'no overview'
data.loc[data.overview.isnull(),'overview'] = 'no overview'

In [100]:
sum(data.overview.isnull())

0

# Popularity

In [101]:
#POPULARITY
data.popularity.values

array([21.946943, 17.015539, 11.7129, ..., '0.903007', '0.003503',
       '0.163015'], dtype=object)

it should be numerical data - it should be changed

In [102]:
data.popularity = data.popularity.astype('float64')

In [103]:
data.popularity.dtype

dtype('float64')

In [104]:
data.popularity.describe()

count    45402.000000
mean         2.922633
std          6.008223
min          0.000000
25%          0.386418
50%          1.128116
75%          3.680504
max        547.488298
Name: popularity, dtype: float64

there is a outlier in popularity becos the max value is 547 hich very greater or extent when compared to min and mean values which are between 2 to 3 values


In [105]:
sum(data.popularity.isnull()) #

3

In [106]:
# 1.Z-SCORE when it greated than 3 and less than -3 then it is outlier(here based on normal distribution
# values is between 3 and -3 we setting same as threshold       )
data['z-score_popularity'] = (data.popularity - data.popularity.mean())/data.popularity.std()



In [107]:
data['z-score_popularity']

0        3.166379
1        2.345603
2        1.463039
3        0.155930
4        0.909568
           ...   
45461   -0.474447
45462   -0.456773
45463   -0.336144
45464   -0.485856
45465   -0.459307
Name: z-score_popularity, Length: 45405, dtype: float64

when we see bove the first data point is greater than 3 and it is outlier

In [108]:
data['outliers'] = (data['z-score_popularity'] > 3) | (data['z-score_popularity'] < -3)

here we are creating new column outliers

In [109]:
data['outliers']

0         True
1        False
2        False
3        False
4        False
         ...  
45461    False
45462    False
45463    False
45464    False
45465    False
Name: outliers, Length: 45405, dtype: bool

To impute or handle the outliers:
1.delete it
OR
2.Replace the outliers UPPER THERSHOLD(q3+(1.5)IQR) to max above outlier values
and Replace the outliers LOWER THRESHOLD(q1-(1.5)IQR)  to min above outliers value

In [110]:
#IQR
q3 = data.popularity.quantile(0.75)
q1 = data.popularity.quantile(0.25)
iqr = q3 - q1

In [111]:
# upper threshold
q3 + (1.5*iqr)

8.621632250000001

In [112]:
# lower threshold
q1 - (1.5*iqr)

-4.554709750000001

In [113]:
# outlier imputation (.CLIP(LOWER THRESHOLD,UPPER THRESHOLD))
data.popularity = data.popularity.clip(-4.55,8.62) #every outlier less than -4.55 will be replaced by -4.55 and
                                                   #every outlier greater than 8.62 will be replaced by 8.62

In [114]:
data.popularity.describe()

count    45402.000000
mean         2.459749
std          2.831169
min          0.000000
25%          0.386418
50%          1.128116
75%          3.680504
max          8.620000
Name: popularity, dtype: float64

now outliers are removed and replaced using IQR

In [115]:
sum(data.popularity.isnull())

3

we can delete this missing values in popularity

In [116]:
data = data.dropna(subset = ['popularity'])

In [ ]:
sum(data.popularity.isnull())

0

now no null value

# Vote count and Vote Average

In [117]:
data.vote_count.values[0]

5415.0

In [118]:
sum(data.vote_count.isnull())

0

In [119]:
data.vote_count.describe()

count    45402.000000
mean       110.000000
std        491.611018
min          0.000000
25%          3.000000
50%         10.000000
75%         34.000000
max      14075.000000
Name: vote_count, dtype: float64

here max value is quite maxi but we have possibility to get votes for particular movie. Therefore this outlier is not gonna affect our data.

# VOTE AVERAGE VALUES

In [120]:
data.vote_average.values[0]

7.7

In [121]:
data.vote_average.value_counts

<bound method IndexOpsMixin.value_counts of 0        7.7
1        6.9
2        6.5
3        6.1
4        5.7
        ... 
45461    4.0
45462    9.0
45463    3.8
45464    0.0
45465    0.0
Name: vote_average, Length: 45402, dtype: float64>

dtype is float - it is correct

In [122]:
sum(data.vote_average.isnull())

0

no null values

In [123]:
data.vote_average.describe()

count    45402.000000
mean         5.619479
std          1.922535
min          0.000000
25%          5.000000
50%          6.000000
75%          6.800000
max         10.000000
Name: vote_average, dtype: float64

#by using vote_count and vote_average we can create a derived column SCORE

In [124]:
data['score'] = (data.vote_average * data.vote_count)

In [125]:
data.score.value_counts

<bound method IndexOpsMixin.value_counts of 0        41695.5
1        16649.7
2          598.0
3          207.4
4          986.1
          ...   
45461        4.0
45462       27.0
45463       22.8
45464        0.0
45465        0.0
Name: score, Length: 45402, dtype: float64>

#Poster path

In [126]:
data.poster_path.value_counts

<bound method IndexOpsMixin.value_counts of 0        /rhIRbceoE9lR4veEXuwCC2wARtG.jpg
1        /vzmL6fP7aPKNKPRTFnZmiUfciyV.jpg
2        /6ksm1sjKMFLbO7UY2i6G1ju9SML.jpg
3        /16XOMpEaLWkrcPqSQqhTmeJuqQl.jpg
4        /e64sOI48hQXyru7naBFyssKFxVd.jpg
                       ...               
45461    /jldsYflnId4tTWPx8es3uzsB1I8.jpg
45462    /xZkmxsNmYXJbKVsTRLLx3pqGHx7.jpg
45463    /d5bX92nDsISNhu3ZT69uHwmfCGw.jpg
45464    /aorBPO7ak8e8iJKT5OcqYxU3jlK.jpg
45465    /s5UkZt6NTsrS7ZF0Rh8nzupRlIU.jpg
Name: poster_path, Length: 45402, dtype: object>

It is string defined correctly but for our use case we know need this column so we are deleting

But in use case if we need this poster path (i.e) iage path of the movie
we should chec whether there is null values
and check for the image in particular link of the path

In [127]:
sum(data.poster_path.isnull())

378

In [128]:
# to find whether we have image in that link or not
# import os #operating syste os package
# os.mkdir("test_dir")  # TO make a directory
# os.path.isfile("/content/movies_metadata.csv") # to check whether the file is present or not


In [129]:
# as for now we dropping the column
data = data.drop(columns =  ['poster_path'])

In [130]:
data.columns

Index(['adult', 'budget', 'id', 'imdb_id', 'original_language',
       'original_title', 'overview', 'popularity', 'production_companies',
       'production_countries', 'release_date', 'revenue', 'runtime',
       'spoken_languages', 'status', 'tagline', 'video', 'vote_average',
       'vote_count', 'CollectionName', 'History', 'Mystery', 'Animation',
       'Fantasy', 'Drama', 'Romance', 'Music', 'Foreign', 'Family',
       'Science Fiction', 'Western', 'Crime', 'Action', 'TV Movie', 'Comedy',
       'War', 'Horror', 'Thriller', 'Documentary', 'Adventure',
       'homePagePresent', 'z-score_popularity', 'outliers', 'score'],
      dtype='object')

Production Companies

In [131]:
data.production_companies.values

array(["[{'name': 'Pixar Animation Studios', 'id': 3}]",
       "[{'name': 'TriStar Pictures', 'id': 559}, {'name': 'Teitler Film', 'id': 2550}, {'name': 'Interscope Communications', 'id': 10201}]",
       "[{'name': 'Warner Bros.', 'id': 6194}, {'name': 'Lancaster Gate', 'id': 19464}]",
       ..., "[{'name': 'American World Pictures', 'id': 6165}]",
       "[{'name': 'Yermoliev', 'id': 88753}]", '[]'], dtype=object)

it is read as string but it should be list of companies

In [132]:
data.production_companies.values[0]

"[{'name': 'Pixar Animation Studios', 'id': 3}]"

In [133]:
sum(data.production_companies.isnull())

0

In [134]:
# TO EXRACT PRODUCER1
import ast
index=0
prod1=[]

for prod in data.production_companies.values: #looping through all rows
  prodlist= ast.literal_eval(prod) #string -> list of dicts
  if len(prodlist) > index:        #if they have atleast 1 producer
    prod1.append(prodlist[index]['name'])
  else:
    prod1.append('No Producer')

data['Producer1'] = prod1


In [139]:
# for extracting producer2
import ast
index = 1
prod2 = []

for prod in data.production_companies.values:
  prodlist = ast.literal_eval(prod)
  if len(prodlist) > index:
    prod2.append(prodlist[index]['name'])
  else:
    prod2.append('No Producer')

data['Producer2'] = prod2

In [142]:
data.Producer2

0                No Producer
1               Teitler Film
2             Lancaster Gate
3                No Producer
4        Touchstone Pictures
                ...         
45461            No Producer
45462            No Producer
45463            No Producer
45464            No Producer
45465            No Producer
Name: Producer2, Length: 45402, dtype: object

In [144]:
data = data.drop(columns = ['production_companies'])

In [146]:
data['Producer1'] = data['Producer1'].str.lower()
data['Producer2'] = data['Producer2'].str.lower()
data['Producer1'] = data['Producer1'].str.strip()
data['Producer2'] = data['Producer2'].str.strip()


In [147]:
data

adult      budget      id    imdb_id original_language  \
0        0.0  30000000.0     862  tt0114709                en   
1        0.0  65000000.0    8844  tt0113497                en   
2        0.0         0.0   15602  tt0113228                en   
3        0.0  16000000.0   31357  tt0114885                en   
4        0.0         0.0   11862  tt0113041                en   
...      ...         ...     ...        ...               ...   
45461    0.0         0.0  439050  tt6209470                fa   
45462    0.0         0.0  111109  tt2028550                tl   
45463    0.0         0.0   67758  tt0303758                en   
45464    0.0         0.0  227506  tt0008536                en   
45465    0.0         0.0  461257  tt6980792                en   

                    original_title  \
0                        toy story   
1                          jumanji   
2                 grumpier old men   
3                waiting to exhale   
4      father of the bride part ii   
...                            ...   
45461                      رگ خواب   
45462          siglo ng pagluluwal   
45463                     betrayal   
45464          satana likuyushchiy   
45465                     queerama   

                                                overview  popularity  \
0      led by woody, andy's toys live happily in his ...    8.620000   
1      when siblings judy and peter discover an encha...    8.620000   
2      a family wedding reignites the ancient feud be...    8.620000   
3      cheated on, mistreated and stepped on, the wom...    3.859495   
4      just when george banks has recovered from his ...    8.387519   
...                                                  ...         ...   
45461        rising and falling between a man and woman.    0.072051   
45462  an artist struggles to finish his work while a...    0.178241   
45463  when one of her hits goes wrong, a professiona...    0.903007   
45464  in a small town live two brothers, one a minis...    0.003503   
45465  50 years after decriminalisation of homosexual...    0.163015   

                                    production_countries release_date  ...  \
0      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-10-30  ...   
1      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-15  ...   
2      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22  ...   
3      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22  ...   
4      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-02-10  ...   
...                                                  ...          ...  ...   
45461             [{'iso_3166_1': 'IR', 'name': 'Iran'}]          NaN  ...   
45462      [{'iso_3166_1': 'PH', 'name': 'Philippines'}]   2011-11-17  ...   
45463  [{'iso_3166_1': 'US', 'name': 'United States o...   2003-08-01  ...   
45464           [{'iso_3166_1': 'RU', 'name': 'Russia'}]   1917-10-21  ...   
45465   [{'iso_3166_1': 'GB', 'name': 'United Kingdom'}]   2017-06-09  ...   

       Horror  Thriller Documentary Adventure homePagePresent  \
0         0.0       0.0         0.0       0.0               1   
1         0.0       0.0         0.0       1.0               0   
2         0.0       0.0         0.0       0.0               0   
3         0.0       0.0         0.0       0.0               0   
4         0.0       0.0         0.0       0.0               0   
...       ...       ...         ...       ...             ...   
45461     0.0       0.0         0.0       0.0               1   
45462     0.0       0.0         0.0       0.0               0   
45463     NaN       NaN         NaN       NaN               0   
45464     NaN       NaN         NaN       NaN               0   
45465     NaN       NaN         NaN       NaN               0   

      z-score_popularity  outliers    score  \
0               3.166379      True  41695.5   
1               2.345603     False  16649.7   
2               1.463039     False    598.0  

#Release Date

In [148]:
data.release_date.values

array(['1995-10-30', '1995-12-15', '1995-12-22', ..., '2003-08-01',
       '1917-10-21', '2017-06-09'], dtype=object)

here datatype is tring but it should be date time datatype

In [149]:
data.release_date.values[0]

'1995-10-30'

In [150]:
sum(data.release_date.isnull())

81

In [151]:
data = data.dropna(subset=['release_date'])

In [152]:
sum(data.release_date.isnull())

0

In [166]:
#convert to date
import datetime
dateString = '2023/September/19'
date = datetime.datetime.strptime(dateString,"%Y/%B/%d")  # str to date time
#datetime.datetime.strftime((date + datetime.timedelta(days = 2, hours = 2)),"%d.%b.%y.%H:%M:%S")  # convert date time to string
date
#date +  datetime.timedelta(days=15)
#date.isocalendar()[1]
#date.hour



datetime.datetime(2023, 9, 19, 0, 0)

In [167]:
# CONVERTING COLUMN RELEASE DATE TO DATE FORMAT YYYY-MM-DD
data.release_date = pd.to_datetime(data.release_date,format='%Y-%m-%d')

<ipython-input-167-5b9edd4581bd>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data.release_date = pd.to_datetime(data.release_date,format='%Y-%m-%d')


In [168]:
data.release_date

0       1995-10-30
1       1995-12-15
2       1995-12-22
3       1995-12-22
4       1995-02-10
           ...    
45460   1991-05-13
45462   2011-11-17
45463   2003-08-01
45464   1917-10-21
45465   2017-06-09
Name: release_date, Length: 45321, dtype: datetime64[ns]

In [169]:
data.columns

Index(['adult', 'budget', 'id', 'imdb_id', 'original_language',
       'original_title', 'overview', 'popularity', 'production_countries',
       'release_date', 'revenue', 'runtime', 'spoken_languages', 'status',
       'tagline', 'video', 'vote_average', 'vote_count', 'CollectionName',
       'History', 'Mystery', 'Animation', 'Fantasy', 'Drama', 'Romance',
       'Music', 'Foreign', 'Family', 'Science Fiction', 'Western', 'Crime',
       'Action', 'TV Movie', 'Comedy', 'War', 'Horror', 'Thriller',
       'Documentary', 'Adventure', 'homePagePresent', 'z-score_popularity',
       'outliers', 'score', 'Producer1', 'Producer2'],
      dtype='object')

# REVENUE

In [170]:
data.revenue.values[0]

373554033.0

In [173]:
data.revenue.values

array([3.73554033e+08, 2.62797249e+08, 0.00000000e+00, ...,
       0.00000000e+00, 0.00000000e+00, 0.00000000e+00])

Revenue can be integer value because float value will not make any sense

In [174]:
data.revenue = data.revenue.astype('int64')

<ipython-input-174-63c4cf273d1f>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data.revenue = data.revenue.astype('int64')


In [178]:
data.revenue.value_counts  # by converting to int the values will be round off

<bound method IndexOpsMixin.value_counts of 0        373554033
1        262797249
2                0
3         81452156
4         76578911
           ...    
45460            0
45462            0
45463            0
45464            0
45465            0
Name: revenue, Length: 45321, dtype: int64>

In [177]:
data.revenue.describe()

count    4.532100e+04
mean     1.123985e+07
std      6.442712e+07
min      0.000000e+00
25%      0.000000e+00
50%      0.000000e+00
75%      0.000000e+00
max      2.787965e+09
Name: revenue, dtype: float64

here 75% value is zero normally revenue will not be zero
so here we have MISSING VALUES

In [179]:
sum(data.revenue == 0)*100/data.shape[0]

83.67864786743452

83% data are missing
handling:
1. fetch or calculate the values from different column
2. delete the 83% missing data


As of now we are keeping as such later with problem stmt we will decide

# RUNTIME

In [180]:
data.runtime.values

array([ 81., 104., 101., ...,  90.,  87.,  75.])

In [181]:
data.runtime.values[0]

81.0

In [182]:
data.runtime.describe()

count    45077.000000
mean        94.193624
std         38.337405
min          0.000000
25%         85.000000
50%         95.000000
75%        107.000000
max       1256.000000
Name: runtime, dtype: float64

In [184]:
sum(data.runtime == 0)

1531

here Movie runtime is zero for 1531 rows but we don't have zero runtime for movies
there are 1531 records so we try imputing the values

In [185]:
data.loc[data.runtime == 0 , 'runtime'] = 95

here it is numerical value , we can go for mean or median
so we are choosing median

In [186]:
sum(data.runtime > 300) # runtime greater than 5 hours

107

In [187]:
data.loc[data.runtime>300,:]

adult       budget      id    imdb_id original_language  \
6747     0.0          0.0   42044  tt0090015                en   
7415     0.0          0.0  140207  tt0080297                en   
7420     0.0          0.0   11659  tt0346336                it   
7933     0.0          0.0   36627  tt0072021                pl   
8635     0.0  100000000.0   29266  tt0063794                ru   
...      ...          ...     ...        ...               ...   
44176    0.0          0.0  444623  tt0366028                ru   
44464    0.0          0.0  197854  tt0143353                ja   
44721    0.0          0.0  458298  tt6792200                en   
44758    0.0          0.0  128733  tt0080230                en   
45462    0.0          0.0  111109  tt2028550                tl   

                  original_title  \
6747                       shoah   
7415   tinker tailor soldier spy   
7420          la meglio gioventú   
7933                       potop   
8635                 война и мир   
...                          ...   
44176                      idiot   
44464                   強殖装甲ガイバー   
44721                the keepers   
44758                  hollywood   
45462        siglo ng pagluluwal   

                                                overview  popularity  \
6747   claude lanzmann directed this 9 1/2 hour docum...    1.978031   
7415   in the bleak days of the cold war, espionage v...    1.564586   
7420   spanning four decades, from the chaotic 1960s ...    6.040001   
7933   following with fire and sword', this is an ada...    0.868637   
8635   seven-hour epic based on the eponymous book by...    2.604571   
...                                                  ...         ...   
44176             tv miniseries of the dostoevsky novel.    0.074146   
44464  sho and his friend tetsurou stumble upon an od...    0.025827   
44721  a seven-part docuseries about the unsolved mur...    1.576170   
44758  the definitive documentary about the american ...    0.055713   
45462  an artist struggles to finish his work while a...    0.178241   

                                    production_countries release_date  ...  \
6747            [{'iso_3166_1': 'FR', 'name': 'France'}]   1985-11-01  ...   
7415    [{'iso_3166_1': 'GB', 'name': 'United Kingdom'}]   1979-09-10  ...   
7420             [{'iso_3166_1': 'IT', 'name': 'Italy'}]   2003-05-19  ...   
7933            [{'iso_3166_1': 'PL', 'name': 'Poland'}]   1974-09-02  ...   
8635            [{'iso_3166_1': 'RU', 'name': 'Russia'}]   1966-03-14  ...   
...                                                  ...          ...  ...   
44176                                                 []   2003-04-01  ...   
44464            [{'iso_3166_1': 'JP', 'name': 'Japan'}]   1989-09-25  ...   
44721  [{'iso_3166_1': 'US', 'name': 'United States o...   2017-05-19  ...   
44758                                                 []   1980-02-09  ...   
45462      [{'iso_3166_1': 'PH', 'name': 'Philippines'}]   2011-11-17  ...   

       Horror  Thriller Documentary Adventure homePagePresent  \
6747      0.0       0.0         1.0       0.0               0   
7415      0.0       1.0         0.0       0.0               0   
7420      0.0       0.0         0.0       0.0               0   
7933      0.0       0.0         0.0       1.0               0   
8635      0.0       0.0         0.0       0.0               0   
...       ...       ...         ...       ...             ...   
44176     1.0       1.0         0.0       0.0               0   
44464     1.0       1.0         0.0       1.0               0   
44721     0.0       0.0         0.0       0.0               1   
44758     1.0       1.0         0.0       0.0               0   
45462     0.0       0.0         0.0       0.0               0   

      z-score_popularity  outliers  score  \
6747           -0.157218     False  313.2   
7415           -0.226031     False   76.0   
7420            0.518850     False  704.0   
7933           -0.

Here the series movies also in the list so there is chance of having 5 hours and more runtime

In [190]:
data.runtime.isnull().sum()

244

In [191]:
data.runtime = data.runtime.fillna(data.runtime.median())

<ipython-input-191-9acad15195a8>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data.runtime = data.runtime.fillna(data.runtime.median())


here filling null values with median values

In [192]:
data.runtime.isnull().sum()

0

# spoken language

In [193]:
data.spoken_languages.values

array(["[{'iso_639_1': 'en', 'name': 'English'}]",
       "[{'iso_639_1': 'en', 'name': 'English'}, {'iso_639_1': 'fr', 'name': 'Français'}]",
       "[{'iso_639_1': 'en', 'name': 'English'}]", ...,
       "[{'iso_639_1': 'en', 'name': 'English'}]", '[]',
       "[{'iso_639_1': 'en', 'name': 'English'}]"], dtype=object)

here it is list of dictonary

In [194]:
data.spoken_languages.values[0]

"[{'iso_639_1': 'en', 'name': 'English'}]"

In [196]:
import ast
totallanguages = []

for prod in data.spoken_languages.values:
  prodlist = ast.literal_eval(prod)
  totallanguages.append(len(prodlist))

data['totallanguages'] = totallanguages

<ipython-input-196-455eda04ee8c>:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['totallanguages'] = totallanguages


In [197]:
data.totallanguages

0        1
1        2
2        1
3        1
4        1
        ..
45460    1
45462    1
45463    1
45464    0
45465    1
Name: totallanguages, Length: 45321, dtype: int64

In [198]:
data = data.drop(columns = ['spoken_languages'])

In [200]:
data.columns

Index(['adult', 'budget', 'id', 'imdb_id', 'original_language',
       'original_title', 'overview', 'popularity', 'production_countries',
       'release_date', 'revenue', 'runtime', 'status', 'tagline', 'video',
       'vote_average', 'vote_count', 'CollectionName', 'History', 'Mystery',
       'Animation', 'Fantasy', 'Drama', 'Romance', 'Music', 'Foreign',
       'Family', 'Science Fiction', 'Western', 'Crime', 'Action', 'TV Movie',
       'Comedy', 'War', 'Horror', 'Thriller', 'Documentary', 'Adventure',
       'homePagePresent', 'z-score_popularity', 'outliers', 'score',
       'Producer1', 'Producer2', 'totallanguages'],
      dtype='object')

# Status

In [203]:
data.status.values

array(['Released', 'Released', 'Released', ..., 'Released', 'Released',
       'Released'], dtype=object)

In [208]:
data.status.value_counts()

Released           44885
Rumored              226
Post Production       97
In Production         19
Planned               13
Canceled               1
Name: status, dtype: int64

In [206]:
sum(data.status.isnull())

80

In [209]:
data = data.dropna(subset=['status'])

In [210]:
sum(data.status.isnull())

0

In [211]:
data.status.value_counts()

Released           44885
Rumored              226
Post Production       97
In Production         19
Planned               13
Canceled               1
Name: status, dtype: int64

# OTHER COLUMNS

In [213]:
data.production_countries

0        [{'iso_3166_1': 'US', 'name': 'United States o...
1        [{'iso_3166_1': 'US', 'name': 'United States o...
2        [{'iso_3166_1': 'US', 'name': 'United States o...
3        [{'iso_3166_1': 'US', 'name': 'United States o...
4        [{'iso_3166_1': 'US', 'name': 'United States o...
                               ...                        
45460    [{'iso_3166_1': 'CA', 'name': 'Canada'}, {'iso...
45462        [{'iso_3166_1': 'PH', 'name': 'Philippines'}]
45463    [{'iso_3166_1': 'US', 'name': 'United States o...
45464             [{'iso_3166_1': 'RU', 'name': 'Russia'}]
45465     [{'iso_3166_1': 'GB', 'name': 'United Kingdom'}]
Name: production_countries, Length: 45241, dtype: object

In [214]:
data = data.drop(columns = ['production_countries'])

In [218]:
data = data.drop(columns = ['video','tagline'])

# FINAL CLEANED DATA

In [223]:
data = data.dropna()  #cleanes if any null values are there

In [224]:
data

adult      budget      id    imdb_id original_language  \
0        0.0  30000000.0     862  tt0114709                en   
1        0.0  65000000.0    8844  tt0113497                en   
2        0.0         0.0   15602  tt0113228                en   
3        0.0  16000000.0   31357  tt0114885                en   
4        0.0         0.0   11862  tt0113041                en   
...      ...         ...     ...        ...               ...   
45457    0.0         0.0  390959  tt0265736                en   
45458    0.0         0.0  289923  tt0252966                en   
45459    0.0         0.0  222848  tt0112613                en   
45460    0.0         0.0   30840  tt0102797                en   
45462    0.0         0.0  111109  tt2028550                tl   

                    original_title  \
0                        toy story   
1                          jumanji   
2                 grumpier old men   
3                waiting to exhale   
4      father of the bride part ii   
...                            ...   
45457    shadow of the blair witch   
45458          the burkittsville 7   
45459              caged heat 3000   
45460                   robin hood   
45462          siglo ng pagluluwal   

                                                overview  popularity  \
0      led by woody, andy's toys live happily in his ...    8.620000   
1      when siblings judy and peter discover an encha...    8.620000   
2      a family wedding reignites the ancient feud be...    8.620000   
3      cheated on, mistreated and stepped on, the wom...    3.859495   
4      just when george banks has recovered from his ...    8.387519   
...                                                  ...         ...   
45457  in this true-crime documentary, we delve into ...    0.076061   
45458  a film archivist revisits the story of rustin ...    0.386450   
45459  it's the year 3000 ad. the world's most danger...    0.661558   
45460  yet another version of the classic epic, with ...    5.683753   
45462  an artist struggles to finish his work while a...    0.178241   

      release_date    revenue  ...  Thriller Documentary  Adventure  \
0       1995-10-30  373554033  ...       0.0         0.0        0.0   
1       1995-12-15  262797249  ...       0.0         0.0        1.0   
2       1995-12-22          0  ...       0.0         0.0        0.0   
3       1995-12-22   81452156  ...       0.0         0.0        0.0   
4       1995-02-10   76578911  ...       0.0         0.0        0.0   
...            ...        ...  ...       ...         ...        ...   
45457   2000-10-22          0  ...       0.0         0.0        0.0   
45458   2000-10-03          0  ...       0.0         0.0        0.0   
45459   1995-01-01          0  ...       0.0         0.0        0.0   
45460   1991-05-13          0  ...       1.0         0.0        0.0   
45462   2011-11-17          0  ...       0.0         0.0        0.0   

       homePagePresent z-score_popularity  outliers    score  \
0                    1           3.166379      True  41695.5   
1                    0           2.345603     False  16649.7   
2                    0           1.463039     False    598.0   
3                    0           0.155930     False    207.4   
4                    0           0.909568     False    986.1   
...                ...                ...       ...      ...   
45457                0          -0.473779     False     14.0   
45458                0          -0.422119     False      7.0   
45459                0          -0.376330     False      3.5   
45460                0           0.459557     False    148.2   
45462                0          -0.456773     False     27.0   

                                    Producer1            Producer2  \
0                     pixar animation studios          no producer   
1                            tristar pictures         teitler film   
2                                warner bros.       lancaster gate   
3 

In [225]:
data.columns

Index(['adult', 'budget', 'id', 'imdb_id', 'original_language',
       'original_title', 'overview', 'popularity', 'release_date', 'revenue',
       'runtime', 'status', 'vote_average', 'vote_count', 'CollectionName',
       'History', 'Mystery', 'Animation', 'Fantasy', 'Drama', 'Romance',
       'Music', 'Foreign', 'Family', 'Science Fiction', 'Western', 'Crime',
       'Action', 'TV Movie', 'Comedy', 'War', 'Horror', 'Thriller',
       'Documentary', 'Adventure', 'homePagePresent', 'z-score_popularity',
       'outliers', 'score', 'Producer1', 'Producer2', 'totallanguages'],
      dtype='object')

In [226]:
data.shape

(45238, 42)

# CREATING ANOTHER DATAFRAME FOR REVENUE

In [227]:
dataBudgetRevenue = data.loc[(data.budget != 0) & (data.revenue != 0),:]
dataBudgetRevenue.shape

(5375, 42)

In [229]:
dataBudgetRevenue.head()

adult      budget     id    imdb_id original_language     original_title  \
0    0.0  30000000.0    862  tt0114709                en          toy story   
1    0.0  65000000.0   8844  tt0113497                en            jumanji   
3    0.0  16000000.0  31357  tt0114885                en  waiting to exhale   
5    0.0  60000000.0    949  tt0113277                en               heat   
8    0.0  35000000.0   9091  tt0114576                en       sudden death   

                                            overview  popularity release_date  \
0  led by woody, andy's toys live happily in his ...    8.620000   1995-10-30   
1  when siblings judy and peter discover an encha...    8.620000   1995-12-15   
3  cheated on, mistreated and stepped on, the wom...    3.859495   1995-12-22   
5  obsessive master thief, neil mccauley leads a ...    8.620000   1995-12-15   
8  international action superstar jean claude van...    5.231580   1995-12-22   

     revenue  ...  Thriller Documentary  Adventure  homePagePresent  \
0  373554033  ...       0.0         0.0        0.0                1   
1  262797249  ...       0.0         0.0        1.0                0   
3   81452156  ...       0.0         0.0        0.0                0   
5  187436818  ...       1.0         0.0        0.0                0   
8   64350171  ...       1.0         0.0        1.0                0   

  z-score_popularity  outliers    score  \
0           3.166379      True  41695.5   
1           2.345603     False  16649.7   
3           0.155930     False    207.4   
5           2.496960     False  14522.2   
8           0.384298     False    957.0   

                                Producer1               Producer2  \
0                 pixar animation studios             no producer   
1                        tristar pictures            teitler film   
3  twentieth century fox film corporation             no producer   
5                     regency enterprises            forward pass   
8                      universal pictures  imperial entertainment   

   totallanguages  
0               1  
1               2  
3               1  
5               2  
8               1  

[5 rows x 42 columns]

In [231]:
# derived column profit
dataBudgetRevenue['profit'] = (dataBudgetRevenue.revenue - dataBudgetRevenue.budget)/dataBudgetRevenue.budget
dataBudgetRevenue['profit'] = dataBudgetRevenue['profit'].clip(-1,50)

<ipython-input-231-77d52f3984e0>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dataBudgetRevenue['profit'] = (dataBudgetRevenue.revenue - dataBudgetRevenue.budget)/dataBudgetRevenue.budget
<ipython-input-231-77d52f3984e0>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dataBudgetRevenue['profit'] = dataBudgetRevenue['profit'].clip(-1,50)


PROFIT = (REVENUE - BUDGET)/BUDGET

In [234]:
dataBudgetRevenue.to_csv('finalCleanedData.csv')  # moving dataframe to csv file

By using above finalcleaned data we need to plot graph for the below questions
1. Do the movies that have high vote_average make ot of profits?
2. Whta genres make the most money?
3. Which producers generall do well?
4. Is there a bias with producers in terms of genres that they do - do specific producer only produce specific genre movies?
5. Will a longer run time bring higher profits?
6. Does the presence of homepage indicate higher profits
7. Which languages ranks in high profit
8. Over period of time,how have the profitability changed - over the period of time,which gnre of movies have grossed well?
9. If the movie speaks multiple languages,is the profit % higher?
